### PR TITLE
WRO-739: Update 4.5 migration guide

### PR DESCRIPTION
**Pull request by MikyungKim**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [ ] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
Added 4.5 migration guide

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)

### Links
[//]: # (Related issues, references)
WRO-739

### Comments
Enact-DCO-1.0-Signed-off-by: Mikyung Kim (mikyung27.kim@lge.com)

### CODE REVIEW

File: docs/migration/enact/migrating-to-enact-4.5.md [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+Developers should ensure their code does not rely on features that are no longer available in these versions.
+
+Please do not hesitate to replace `enzyme` with [@testing-library/react](https://testing-library.com/docs/react-testing-library/intro/) when you make your unit tests.
+
+As `@enact/cli` updates `react` and `react-dom` to `18.x`, `PrerenderPlugin` will automatically
+convert `ReactDOMClient.createRoot` to `ReactDOMClient.hydrateRoot` for prerendered apps.  
+Please make sure to follow the above new `createRoot` API pattern to work prerendering properly.
+
+As we update to `eslint 8`, some of the lint rules could be changed. If you run into unknown lint warnings or errors, don't be afraid, and please proceed to fix them. They are likely to be the rules from [eslint-plugin-react](https://github.com/yannickcr/eslint-plugin-react), so refer to the console message and look up which rule is related.  
+The in-editor-linting will work just fine if you followed the guide from migration to 4.0.
```
Comment (seunghoh) [inline docs/migration/enact/migrating-to-enact-4.5.md:165]: What do you mean exactly? `if you followed the guide from migration to 4.0`?

File: docs/migration/enact/migrating-to-enact-4.5.md [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+Please do not hesitate to replace `enzyme` with [@testing-library/react](https://testing-library.com/docs/react-testing-library/intro/) when you make your unit tests.
+
+As `@enact/cli` updates `react` and `react-dom` to `18.x`, `PrerenderPlugin` will automatically
+convert `ReactDOMClient.createRoot` to `ReactDOMClient.hydrateRoot` for prerendered apps.  
+Please make sure to follow the above new `createRoot` API pattern to work prerendering properly.
+
+As we update to `eslint 8`, some of the lint rules could be changed. If you run into unknown lint warnings or errors, don't be afraid, and please proceed to fix them. They are likely to be the rules from [eslint-plugin-react](https://github.com/yannickcr/eslint-plugin-react), so refer to the console message and look up which rule is related.  
+The in-editor-linting will work just fine if you followed the guide from migration to 4.0.  
+Don't forget to install ESlint globally and uninstall any previous globally-installed Enact linting package.
+
```
Comment (seunghoh) [inline docs/migration/enact/migrating-to-enact-4.5.md:167]: how about adding example like `npm install -g eslint`

**Review by seunghoh — APPROVED** [review]
Good work! 
I left some questions.

File: docs/migration/enact/migrating-to-enact-4.5.md [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+## Overview
+This document lists changes between Enact versions 4.0.x and 4.5 likely to affect most apps.  If you
+are coming from Enact 3.x, please [migrate to 4.0](./migrating-to-enact-4.md) and then consult
+this guide.
+
+## General Changes
+
+### React and React DOM
+Enact 4.5 updates the `react` and `react-dom` dependencies to 18.x.  Developers should ensure
+their code does not rely on features that are no longer available in these versions.
```
[the remark this answers is not in the document]
Comment (MikyungKim) [reply docs/migration/enact/migrating-to-enact-4.5.md:15]: 🤔 Maybe we don't need this.

File: docs/migration/enact/migrating-to-enact-4.5.md (hunk 1 above)
Comment (MikyungKim) [reply docs/migration/enact/migrating-to-enact-4.5.md:162]: I meant you will be fine if you've done works to make in-editor-linting before.
But I just realized we haven't mentioned this in the migration guides before.
I must have been confused with the internal developer guide. Maybe we need to remove the in-editor-linting part or write it more in detail.

File: docs/migration/enact/migrating-to-enact-4.5.md (hunk 2 above)
Comment (MikyungKim) [reply docs/migration/enact/migrating-to-enact-4.5.md:164]: 🆗

**Pull request merged**